test/ZendRouterTest.php

```diff
@@ -424,4 +425,23 @@ public function testRoutesCanMatchImplicitHeadAndOptionsRequests($method)
         $this->assertTrue($result->isSuccess());
         $this->assertSame($route, $result->getMatchedRoute());
     }
+
+    public function testUriGenerationMayUseOptions()
```

Comment:
Not sure how to go about that, which may indicate that the recursive merging isn't necessary yet. Currently, the only options I could find to test were around the translator; all others are ignored currently, as they would require using the zend-http URI generation; this implementation takes care of the other options known to Segment routes (query string argument and fragment support).

I'll remove the `ArrayUtils::merge` usage for now, and instead use `array_replace_recursive()`.